cpp/ycm/tests/CMakeLists.txt

```diff
@@ -33,6 +33,11 @@ if ( USE_SYSTEM_GMOCK )
   find_package( GTest REQUIRED )
   find_package( GMock REQUIRED )
 else()
+  if ( WIN32 )
+    # Override BUILD_SHARED_LIBS option in gmock and gtest CMakeLists
+    set( BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries (DLLs)." )
```

Comment:
There is a difference with the Flightcrew project: the `ycm_core` and `ycm_client_support` targets are built as shared libraries. If we want to link statically `ycm_core_tests`, we also need the static version of those libraries.